Zend/zend_API.h

```diff
@@ -1124,20 +1124,23 @@ static zend_always_inline zval *zend_try_array_init(zval *zv)
 #define FAST_ZPP 1
 
 #define Z_EXPECTED_TYPES(_) \
-	_(Z_EXPECTED_LONG,		"int") \
-	_(Z_EXPECTED_BOOL,		"bool") \
-	_(Z_EXPECTED_STRING,	"string") \
-	_(Z_EXPECTED_ARRAY,		"array") \
-	_(Z_EXPECTED_FUNC,		"valid callback") \
-	_(Z_EXPECTED_RESOURCE,	"resource") \
-	_(Z_EXPECTED_PATH,		"a valid path") \
-	_(Z_EXPECTED_OBJECT,	"object") \
-	_(Z_EXPECTED_DOUBLE,	"float") \
-	_(Z_EXPECTED_NUMBER,	"int or float") \
-	_(Z_EXPECTED_STRING_OR_ARRAY, "string or array") \
-
-#define Z_EXPECTED_TYPE_ENUM(id, str) id,
-#define Z_EXPECTED_TYPE_STR(id, str)  str,
+	_(Z_EXPECTED_LONG,		"int", "of type ") \
```

Comment:
I'm not sure at all that this is the cleanest solution to be able to form an error message. Maybe a 0/1 would be enough for differentiating the two cases? Or I should do something completely different?

Reply:
Is the type name used separately from the prefix? That is, why can't they be combined into one string?

Reply:
I didn't want to do that because as it's zend API, I was afraid that something outside php-src uses it. So I shouldn't worry about this then?

Reply:
Nah, this is fine to change.